### PR TITLE
feat: generate token description

### LINKED ISSUE #2107: Tokens: Generate description if empty
Issue created from #1926. 
As a user who is creating token using custom api option, when i select list of resources and permissions, and click `Generate`, if I do not provide the  description, the description is generated in the backend based on their selections but NOT left empty.

Generate field name:
{READ}{WRITE}{RESOURCE_NAME}{RESOURCE_SUBNAME} {repeat for all resources}

### PR DESCRIPTION
**Pull request by blegesse-git**
Closes #2107 

When user who is creating token using custom api option, selects list of resources and permissions, and clicks Generate, if  token description is not provided, the description is generated in the backend based on user selections. 

In the case of user giving a token multiple permissions, a token description with `Read Multiple Write Multiple` is generated.

https://user-images.githubusercontent.com/66275100/134956806-267961b9-0516-4525-b70f-1149adf4e722.mov

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 54 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2710)

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ import {getResourcesStatus} from 'src/resources/selectors/getResourcesStatus'
 import {formatApiPermissions} from 'src/authorizations/utils/permissions'
 
 import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+import {capitalize} from 'lodash'
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:44]: nit: lodash is a library, can we move this import up to all the library imports

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx [hunk 2]
```diff
@@ -147,13 +148,52 @@ const CustomApiTokenOverlay: FC<Props> = props => {
     setPermissions(newPerm)
   }
 
+  const generateDescription = apiPermissions => {
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:151]: Can we move this function out to into utils folder?

File: src/authorizations/components/redesigned/TokenRow.tsx [hunk 3]
```diff
@@ -72,7 +68,6 @@ class TokensRow extends PureComponent<Props> {
             onUpdate={this.handleUpdateName}
             onClick={this.handleClickDescription}
             name={description}
-            noNameString={DEFAULT_TOKEN_DESCRIPTION}
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/TokenRow.tsx:75]: IMO, we still need to keep this here. Our users may have tokens without any description from before this change and for that token we still want to display the default token description.

**Review by Palakp41 — CHANGES_REQUESTED** [review]
Everything looks great, just some small suggestions. Great work on this! 🎉

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx (hunk 1 above)
Comment (blegesse-git) [reply src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:44]: Absolutely!

File: src/authorizations/components/TokenRow.tsx [hunk 4]
```diff
@@ -45,6 +45,7 @@ class TokenRow extends PureComponent<Props> {
     const {auth} = this.props
     const labelText = this.isTokenEnabled ? 'Active' : 'Inactive'
     const date = new Date(auth.createdAt)
+
```
Comment (hoorayimhelping) [inline src/authorizations/components/TokenRow.tsx:48]: 👏 easily the best change i've ever seen

😛😛😛

File: src/authorizations/utils/permissions.ts [hunk 5]
```diff
@@ -269,3 +270,38 @@ export const formatApiPermissions = (permissions, orgID) => {
...
+      } else {
+        generatedDescription += ` ${capitalize(perm.action)} ${
+          perm.resource.type
+        } `
+      }
+    })
+  }
+
+  return generatedDescription
+}
```
Comment (hoorayimhelping) [inline src/authorizations/utils/permissions.ts:307]: let's see a bunch of unit tests for this method!

I'd like you to think about all the different permutations that this can produce and try to write a test case for each type of one. the more cases, the more likely you are to find edge cases and head off any weirdness before users see it.

Let me know if you'd like to pair up on this!

**Review by hoorayimhelping — COMMENTED** [review]
good work, very straightforward and well scoped PR!

I'd like to see a bunch of unit tests for `generateDescription`

**Review by Palakp41 — APPROVED** [review]
Looks good!

File: src/authorizations/components/TokenRow.tsx (hunk 4 above)
Comment (blegesse-git) [reply src/authorizations/components/TokenRow.tsx:48]: lol! I poured blood, sweat, and tears working on line 48.

File: src/authorizations/utils/permissions.test.ts [hunk 6]
```diff
@@ -532,3 +669,31 @@ test('toggleSelectedBucket test', () => {
...
+      'Read telegraf Write telegraf'
+    )
+  })
+
+  test('creates a description for token with less than 2 permissions', () => {
+    expect(generateDescription(apiPermission5)).toBe(
+      'Read buckets _monitoring Write buckets _monitoring'
+    )
+  })
+})
```
Comment (hoorayimhelping) [inline src/authorizations/utils/permissions.test.ts:699]: 👏 👏 👏 👏 👏

**Review by hoorayimhelping — APPROVED** [review]
awesome!
